### PR TITLE
feat(components): [radio-group] add `clearable`

### PR DESCRIPTION
**Pull request by hooray**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

# Description

When radio-group is used in a form, and when it is not a required option, add a `clearable` attribute to it, so that the selected radio can be cleared.

### CODE REVIEW

File: packages/components/radio/src/radio.vue [hunk 1]
```diff
@@ -59,4 +59,10 @@ const { radioRef, radioGroup, focus, size, disabled, modelValue, actualValue } =
 function handleChange() {
   nextTick(() => emit('change', modelValue.value))
 }
+
+function handleClick() {
+  if (radioGroup?.clearable && modelValue.value === actualValue.value) {
+    modelValue.value = undefined
+  }
+}
```
Comment (kooriookami) [inline packages/components/radio/src/radio.vue:67]: Duplicate of the above. It can be removed to `useRadio` hook.

**Review by kooriookami — CHANGES_REQUESTED** [review]

File: packages/components/radio/src/radio.vue (hunk 1 above)
Comment (hooray) [reply packages/components/radio/src/radio.vue]: done

File: packages/components/radio/src/use-radio.ts [hunk 2]
```diff
@@ -58,6 +58,12 @@ export const useRadio = (
     computed(() => isGroup.value && isPropAbsent(props.value))
   )
 
+  function cleared() {
```
Comment (kooriookami) [inline packages/components/radio/src/use-radio.ts:61]:
```suggestion
  function handleClear() {
```
I think `handleClear` is better.

**Review by kooriookami — CHANGES_REQUESTED** [review]

File: packages/components/radio/src/use-radio.ts (hunk 2 above)
Comment (hooray) [reply packages/components/radio/src/use-radio.ts]: It makes sense

**Review by kooriookami — APPROVED** [review]

**Comment by tolking** [comment]
Thank for your contribution.

IMO, This is not a good idea because it does not conform to <input type="radio">

**Comment by hooray** [comment]
> Thank for your contribution.
> 
> IMO, This is not a good idea because it does not conform to <input type="radio">

I'm going to close this pr, because it may not be possible to reach a consensus, I admit that this PR has indeed broken the default behavior of <input type="radio"> , but it is indeed necessary in business scenarios.

So if you have such needs, you can refer to the alternative solutions below

https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGVsLXJhZGlvLWdyb3VwIHYtbW9kZWw9XCJyYWRpb1wiPlxuICAgIDxlbC1yYWRpbyA6dmFsdWU9XCIzXCIgQGNsaWNrLm5hdGl2ZS5wcmV2ZW50PVwiaGFuZGxlQ2xpY2soMylcIj5PcHRpb24gQTwvZWwtcmFkaW8+XG4gICAgPGVsLXJhZGlvIDp2YWx1ZT1cIjZcIiBAY2xpY2submF0aXZlLnByZXZlbnQ9XCJoYW5kbGVDbGljayg2KVwiPk9wdGlvbiBCPC9lbC1yYWRpbz5cbiAgICA8ZWwtcmFkaW8gOnZhbHVlPVwiOVwiIEBjbGljay5uYXRpdmUucHJldmVudD1cImhhbmRsZUNsaWNrKDkpXCI+T3B0aW9uIEM8L2VsLXJhZGlvPlxuICA8L2VsLXJhZGlvLWdyb3VwPlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcblxuY29uc3QgcmFkaW8gPSByZWYoMylcblxuZnVuY3Rpb24gaGFuZGxlQ2xpY2sodmFsKSB7XG4gIHJhZGlvLnZhbHVlID0gdmFsID09PSByYWRpby52YWx1ZSA/ICcnIDogdmFsXG59XG48L3NjcmlwdD5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIl9vIjp7fX0=

**Pull request closed without merging**